### DISCUSSION TITLE
Problems with GTDB taxonomy assigments to MARdb sequences

### DISCUSSION THREAD

**Discussion by micronuria** [Robaina/MetaTag#42] [General]
I have encounter two problems when using the relabeltree.py --taxonomy option to assign taxonomy to MARdb sequeces:

1) some genomes from the initial MARdb files that Jose sent are missing the MARdb ID and therefore, those sequences cannot be classified with their ID. To solve it, we need to:
          a) First, figure out if all teams are working with the same files (Complete and Partial genomes, QC by Jose)
          b) Check with José if the MAR IDs were removed during the QC he did - I will do that once we figure out the firs part.

2) For those sequences with MAR IDs the script is assigning the NCBI taxonomy and not the GTDB. For example:

![](https://files.slack.com/files-pri/T02E3G0P5DX-F032TC0QAL8/image.png)

One of the genera that appears in my tree classified as the NCBI (Beta) but in GTDB is Gamma.
This happens with several Betas that have been reclassified as Gammas, for example in the tree

 
![](https://files.slack.com/files-pri/T02E3G0P5DX-F0320PRJX5K/image.png)

The label color is the GTDB web classification and the square in the right the taxonomy applied by the script. Green: Zproteobacteria, Pink: Gamma, Orange: Beta
I have another example with a Streptomyces that has the NCBI taxonomy in the tree label and not the GTDB.
Are we are using an older release??

**Comment by micronuria** [comment]
It is related with issues #14 and #31

**Comment by Robaina** [comment]
Hi there,

one of the issues seems to be due to MAR db employing a NCBI taxonomy tool to classify genomes instead of GTDB-TK. Thus, the observed taxonomy in the final output, since we were using the MAR taxonomy database.

Doing some research, I have found that the [Oceans Microbiomics Database](https://microbiomics.io/ocean/) used GTDB-TK to classify their genome collection ([paper here](https://www.biorxiv.org/content/10.1101/2021.03.24.436479v1.full)). So we can use their database to assign taxonomy to MAR db genomes, at least the ones that are included in their genome collection.

I'll expand the codebase to include the possibility to assign NCBI or GTDB taxonomy to MAR sequences (those whose genome contains assigned taxonomy, that is).

This will solve one of the issues. Regarding the lack of MMP identifiers, this is something not related to the code per se, but to the MAR db preprocessing. We will have to dig in to figure out what's happening.

I have opened issue #41 to address this problem

**Comment by micronuria** [comment]
After asking Jose and showing him this two examples:

![imagen](https://user-images.githubusercontent.com/90261939/153036750-7feb38b2-a0ed-4909-b28b-25d56ea59384.png)

Check Cellulophaga_baltica_18 that is missing the ID

![imagen](https://user-images.githubusercontent.com/90261939/153036846-98f4f44a-6428-45ab-b008-8a4fe271cee5.png)

This is a tree, where the cyanos with ids and taxonomy assigned by relabeltree.py has a green square and those that are missing the ID does not have one. 

Here is his answer:
Esa última parte que empieza con MMP es el identificador en mar. Ese identificador está en la hoja de excel donde están los metadatos y ahí se puede encontrar la taxonomía. Añadí ese identificador a los genomas parciales de mar.
Sí, ahí hay duplicados. C. baltica 18 está sin y con identificador, lo mismo que NN016038. Para mí no es problema porque quito los péptidos duplicados. Están duplicados porque esos genomas están tanto en el lado de genomas parciales como en el lado de genomas completos. Los de MAR los han puesto en los dos sitios por alguna razón.
De los synechococcus, se ve que los que no tienen taxonomía en esa lista tienen los genomas completos. Lo que pasa es que el fichero de metadatos es de la nueva versión que tiene menos genomas que en la versión anterior con la que trabajo. Tampoco para mí es un problema porque les asigno la taxonomía de los peptidos que están cerca en el árbol.

So, I need first to figure out about the duplicates and then, we will see what can be done with the IDs...

**Comment by micronuria** [comment]
After a meeting over skype, we have decided to solve this problem as follows:

- Use genome IDs to assign GTDB taxonomy to MAR genomes instead of taxid or specie names. The proportion of MAR genomes that are missing those identifiers is not large:
         complete: 948 / 970
         partial: 10461 / 13237
- Then, to build a common database:
      1 - Filter MAR genomes to keep only those with GTDB taxonomy
      2 - Join filtered MAR db with Paoli and OceanDNA databases
      3 - Preprocess joined dbs, including removal of duplicates
      4 - Share the final fasta so everybody in the project uses the same starting database

We might need to decide how to solve conflicts when removing duplicates, but we will see how many conflicts arise during the preprocess.
